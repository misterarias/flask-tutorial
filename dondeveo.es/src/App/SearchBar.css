.container {
  text-align: center;
  background-color: white;
  padding: 10px;
  margin: 60px;
}

.header {
  height: 60px;
  padding: 10px;
  color: #222;
}

.title {
  font-size: 1.5em;
}

.searchbar input {
  width: 150px;
  height: 40px;
  padding-left: 5px;
  margin: 0 5px;
  font-size: 16px;
}
.searchbar button {
  height: 40px;
  font-size: 16px;
}
