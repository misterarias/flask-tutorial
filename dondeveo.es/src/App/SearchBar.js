import React, { Component } from 'react';
import { Input, Button, Col, Row } from 'reactstrap';

export default class SearchBar extends Component {

  constructor() {
    super();

    this.handleChange = this.handleChange.bind(this);
    this.handleSubmit = this.handleSubmit.bind(this);

    this.state = {
      query: ""
    }
  }

  handleChange(ev) {
    this.setState({
      query: ev.target.value
    });
  }

  handleSubmit(ev) {
    this.props.handleSubmit(this.state.query);
  }

  render() {

    const pushBottomStyle = {
      marginBottom: '1rem'
    }

    return (
      <Row style={pushBottomStyle}>

        <Col sm={10} >
          <Input
            type="text"
            name="searchTerm"
            onChange={this.handleChange}
            placeholder="Título de película..."
          />
        </Col>

        <Col sm={2}>
          <Button
            color="primary"
            onClick={this.handleSubmit}
          > Search </Button>
        </Col>

      </Row>
    )
  }
}
