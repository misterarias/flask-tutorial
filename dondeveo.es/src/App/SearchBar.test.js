'use strict'

import React from 'react';
import SearchBar from './SearchBar';

import sinon from 'sinon' ;

import {configure, mount} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
configure({ adapter: new Adapter() })

import renderer from 'react-test-renderer';

it('renders correctly', () => {
  const tree = renderer.create(
    <SearchBar />
  ).toJSON();

  expect(tree).toMatchSnapshot();
});

test('searchBar starts with no text', () => {
  const searchBar = mount(
    <SearchBar />
  );

  const searchBarText = searchBar.find('input');
  expect(searchBarText.text()).toBe("") ;
});

test('searchBar triggers an event when submit is clicked', () => {

  const mockEvent = sinon.spy()
  const searchBar = mount(
    <SearchBar handleSubmit={mockEvent} />
  );

  searchBar.setState({ query: 'Test me' });
  searchBar.find('button').simulate('click');

  expect(mockEvent.calledOnce).toBeTruthy();
  expect(mockEvent.args[0][0]).toBe('Test me');
});
