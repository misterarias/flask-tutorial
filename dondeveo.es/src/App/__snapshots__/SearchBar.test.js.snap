// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="row"
  style={
    Object {
      "marginBottom": "1rem",
    }
  }
>
  <div
    className="col col-sm-10"
  >
    <input
      className="form-control"
      name="searchTerm"
      onChange={[Function]}
      placeholder="Título de película..."
      type="text"
    />
  </div>
  <div
    className="col col-sm-2"
  >
    <button
      className="btn btn-primary"
      onClick={[Function]}
      type="button"
    >
       Search 
    </button>
  </div>
</div>
`;
